server/cmwell-ws/app/controllers/Application.scala

```diff
@@ -3001,6 +3001,35 @@ callback=< [URL] >
     }
   }
 
+  def handleStpControl(agent: String) = Action.async(parse.raw) { implicit req =>
+
+    isAdminEvenNonProd(req) match {
+      case true => {
+
+        val headers = req.cookies.get("X-CM-WELL-TOKEN2") match {
```

Comment:
Please use `AuthUtils.extractTokenFrom(req)` instead of getting the cookie explicitly.

Reply:
Now redundant as not needed when using crudServiceFS